code/xcvm/lib/core/src/gateway.rs

```diff
@@ -131,20 +151,23 @@ impl AssetReference {
 	pub fn denom(&self) -> String {
 		match self {
 			AssetReference::Native { denom } => denom.clone(),
-			AssetReference::Virtual { cw20_address } => format!("cw20:{}", cw20_address),
+			AssetReference::Virtual { cw20_address } => ["cw20:", cw20_address.as_str()].concat(),
 		}
 	}
 }
 
-#[cw_serde]
-#[derive(QueryResponses)]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+#[cfg_attr(feature = "std", derive(schemars::JsonSchema, QueryResponses))]
```

Comment:
Maybe we can just drop QueryResponses? To be honest I don’t understand what it gives us.

Reply:
see `returns` . It allows to generate nice json schema so that people can generate nice TypeScript code for FE.